server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java

```diff
@@ -721,15 +722,17 @@ private void parseProperties(Node parent, ServiceProperties properties) {
         }
     }
 
-    private Collection<URI> resolveURIs(String[] acceptURIs) throws URISyntaxException {
-        Collection<URI> urisWithPort = new HashSet<>();
+    private Collection<String> resolveURIs(String[] acceptURIs) throws URISyntaxException {
+        Collection<String> urisWithPort = new HashSet<>();
         for (String uri : acceptURIs) {
-            URI resolvedURI = resolveURI(getCanonicalURI(uri, true));
+            // TODO: Add initial version of this to TcpResourceAddressSpi
+            String resolvedURI = /*resolveURI(*/(getCanonicalURI(uri, true)).toString();
```

Comment:
It seems like resolveURI adds default port. For e.g http://localhost/path to http://localhost:80/path
But I am not sure if your change would cause an issue. Also, we can add resolveURI(String)

It seems to me that some of this functionality can go into corresponding ResourceAddressFactorySpi (they can add path and default port). But we will keep the changes small and not go there for now.